kustomize/commands/edit/set/setsecret_test.go

```diff
@@ -0,0 +1,213 @@
+  namespace: test-ns
+  type: Opaque
+`,
+			Args:             []string{"test-secret", "--from-literal=key3=val2"},
+			ExpectedErrorMsg: "key 'key3' not found in resource",
+		},
+		{
+			Name: "fails to set a value because secret doesn't exist",
+			KustomizationFileContent: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- literals:
+  - key1=val1
+  name: test-secret
+  type: Opaque
+- literals:
+  - key2=value
+  name: another-secret
+  namespace: a-namespace
+  type: Opaque
+`,
+			Args:             []string{"test-secret2", "--from-literal=key3=val2"},
+			ExpectedErrorMsg: "unable to find Secret with name '\"test-secret2\"'",
+		},
+		{
+			Name: "fails validation because no attributes are being changed",
+			KustomizationFileContent: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- literals:
+  - a-test-key=a-test-value
+  name: test-secret
+  namespace: test-ns
+  type: Opaque
+`,
+			Args:             []string{"test-secret", "--namespace=test-ns"},
+			ExpectedErrorMsg: "at least one of [--from-literal, --new-namespace] must be specified",
+		},
+		{
+			Name: "fails when a literal source doesn't have a key",
+			KustomizationFileContent: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- literals:
+  - some-key=some-value
+  name: test-secret
+  type: Opaque
+`,
+			Args:             []string{"test-secret", "--from-literal=value"},
+			ExpectedErrorMsg: "literal values must be specified in the key=value format",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			kustomization, err := testutils_test.SetupEditSetConfigMapSecretTest(t, newCmdSetSecret, tc.KustomizationFileContent, tc.Args)
+
+			require.Nil(t, kustomization)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), tc.ExpectedErrorMsg)
+		})
+	}
+}
+
+func TestSuccessCasesEditSetSecret(t *testing.T) {
+	testCases := []testutils_test.SuccessCase{
+		{
+			Name: "set a value successfully",
+			KustomizationFileContent: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- literals:
+  - random-key=random-value
+  - key1=value
+  name: test-secret
+  type: Opaque
+`,
+			ExpectedLiterals: []string{"key1=val2", "random-key=random-value"},
+			Args:             []string{"test-secret", "--from-literal=key1=val2"},
+		},
+		{
+			Name: "successfully update namespace of target secret",
+			KustomizationFileContent: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- literals:
+  - a-key=test
+  - another-key=value
+  name: test-secret
+  namespace: test-ns
+  type: Opaque
+`,
+			Args:              []string{"test-secret", "--namespace=test-ns", "--new-namespace=test-new-ns"},
+			ExpectedNamespace: "test-new-ns",
+		},
+		{
+			Name: "successfully update namespace of target secret with empty namespace in file and namespace specified in command",
```

Comment:
Could you add a document for the help message that we can use the "namespace: default" value when omitting the namespace in kustomization.yaml?

Reply:
Added this in 66ba468! Let me know your thoughts on the new wording 😄

Reply:
Thanks.
So, In my understanding, this command uses`"--namespace=default"` literally to change the namespace for the namespace omit entry. If that is right, it may be a little bit difficult to understand, I think.
Please tell me if I had a misreading. 🙏

Reply:
Hi there, @koba1t! Sorry for the delay in answering!

This test case aims to verify that using `--namespace=default` or omitting the namespace will have the same effect, which is apply the changes to the item in the kustomization file that either doesn't have a namespace specified and therefore is implicitly defined to be targeting the `default` namespace, or to the item that has the `default` namespace specified.

The goal is to have both of these declarations attain the same result, regardless of whether the `--namespace=default` option was specified or omitted:
```yaml
secretGenerator:
- literals:
  - key1=value1
  name: test-secret
  namespace: default
  type: Opaque
```
```yaml
secretGenerator:
- literals:
  - key1=value1
  name: test-secret
  type: Opaque
```

Does this make sense?

Reply:
Sorry that this is confusing, it's a little difficult to express in words 😅

Reply:
I super apologize for confusing you...

> So, In my understanding, this command uses"--namespace=default" literally to change the namespace for the namespace omit entry. If that is right, it may be a little bit difficult to understand, I think.

This comment was targeting a line of the document in the help message.
At https://github.com/kubernetes-sigs/kustomize/pull/5467/files#diff-004c47d68f7d5e3beb4b3f63d8328f2f8331df350dfdf2fc7da586fc1b931799R33.

That comment was not related to this test case; I'm so sorry...

Reply:
Ahhh gotcha! I see what you mean, let me update that text.

Reply:
Updated via 3bb9a6d4147aef835e075847bd2778b0fff38646, let me know your thoughts 🙂

Reply:
Thanks!